File: client/src/app/_models/discount.ts
export class Discount {
    id:number=0
    type:number=0
    deleted:boolean=false
    value :number=0;
    productId :number;
    categoryId :number;
    startDate:Date = new Date();
    endDate :Date = new Date();
    startDateString:string = '';
    endDateString:string = '';

    constructor(){}
}

File: API/DBAccess/Entities/Discount.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.DBAccess.Entities
{
    public class Discount
    {
        public int Id { get; set; }
        [Required]
        public int Type { get; set; }
        [Required]
        public double Value { get; set; }
        public int? ProductId { get; set; }
        public Product Product { get; set; }
        public int? CategoryId { get; set; }
        public Category Category { get; set; }
        public bool Deleted { get; set; }
        [Required]
        public DateTime StartDate { get; set; }
        [Required]
        public DateTime EndDate { get; set; }
        [Required]
        public DateTime ModDate { get; set; }
    }
}